### ISSUE TITLE
some bug in model

### ISSUE THREAD

**Issue by chinakook** [skhcjh231/MATR_codebase#6]
https://github.com/skhcjh231/MATR_codebase/blame/ba05a98d451b3541c1a5377026f17dc1102fa217/models/models.py#L206

```python
memory_queue[i] = torch.cat((b_memory_queue, current_segment[i]), dim=0)[self.n_seglen:]
```
In this statement, the batch axis of b_memory_queue is axis 0, while the batch axis of current_segment is axis 1. Therefore, they cannot be directly concatenated using torch.cat and need to be modified as follows:
```python
memory_queue[i] = torch.cat((b_memory_queue, current_segment[:,i,:]), dim=0)[self.n_seglen:]
```
